File: testdata/jennies/builders/constant_assignment/GoBuilder/sandbox/somestruct_builder_gen.go
package sandbox

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[SomeStruct] = (*SomeStructBuilder)(nil)

type SomeStructBuilder struct {
    internal *SomeStruct
    errors cog.BuildErrors
}

func NewSomeStructBuilder() *SomeStructBuilder {
	resource := NewSomeStruct()
	builder := &SomeStructBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *SomeStructBuilder) Build() (SomeStruct, error) {
	if err := builder.internal.Validate(); err != nil {
		return SomeStruct{}, err
	}
	
	if len(builder.errors) > 0 {
	    return SomeStruct{}, cog.MakeBuildErrors("sandbox.someStruct", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *SomeStructBuilder) Editable() *SomeStructBuilder {
    builder.internal.Editable = true

    return builder
}

func (builder *SomeStructBuilder) Readonly() *SomeStructBuilder {
    builder.internal.Editable = false

    return builder
}

func (builder *SomeStructBuilder) AutoRefresh() *SomeStructBuilder {
            valAutoRefresh := true
    builder.internal.AutoRefresh = &valAutoRefresh

    return builder
}

func (builder *SomeStructBuilder) NoAutoRefresh() *SomeStructBuilder {
            valAutoRefresh := false
    builder.internal.AutoRefresh = &valAutoRefresh

    return builder
}

File: testdata/jennies/builders/constant_assignment/TypescriptBuilder/src/sandbox/someStructBuilder.gen.ts
import * as cog from '../cog';
import * as sandbox from '../sandbox';

export class SomeStructBuilder implements cog.Builder<sandbox.SomeStruct> {
    protected readonly internal: sandbox.SomeStruct;

    constructor() {
        this.internal = sandbox.defaultSomeStruct();
    }

    /**
     * Builds the object.
     */
    build(): sandbox.SomeStruct {
        return this.internal;
    }

    editable(): this {
        this.internal.editable = true;
        return this;
    }

    readonly(): this {
        this.internal.editable = false;
        return this;
    }

    autoRefresh(): this {
        this.internal.autoRefresh = true;
        return this;
    }

    noAutoRefresh(): this {
        this.internal.autoRefresh = false;
        return this;
    }
}

